src/components/Rockets.js

```diff
@@ -0,0 +1,59 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { reserveRocket, cancelReserve } from '../redux/rocket/rocketSlice';
+import './styles/Rocket.css';
+
+const Rockets = () => {
+  const rocket = useSelector((state) => state.rocket.rocket);
+  const dispatch = useDispatch();
+
+  const reserveRocketHandler = (rocketId) => {
+    dispatch(reserveRocket(rocketId));
+  };
+
+  const cancelRocketHandler = (rocketId) => {
+    dispatch(cancelReserve(rocketId));
+  };
+
+  return (
+    <div className="rockets-box">
+      {rocket.map((rocket) => (
+        <div className="rocket" key={rocket.id}>
+          <img
+            src={rocket.image_url}
+            alt={rocket.rocket_name}
+            className="rocket-img"
+          />
+          <div className="rocket-info">
+            <h2 className="rocket-name">{rocket.rocket_name}</h2>
+            <p className="rocket-note">
+              {rocket.reserved && (
+                <span className="px-3 text-[18px] rounded-md mr-2 bg-blue-400 text-white">
+                  Reserved
+                </span>
+              )}
+              {rocket.description}
+            </p>
+            {rocket.reserved ? (
+              <button
+                className="rocket-btn text-gray-500 py-[5px] border-2 border-gray-500"
+                type="button"
+                onClick={() => cancelRocketHandler(rocket.id)}
+              >
+                Cancel Reservation
+              </button>
+            ) : (
+              <button
+                className="rocket-btn bg-blue-500 text-white py-[5px]"
+                type="button"
+                onClick={() => reserveRocketHandler(rocket.id)}
+              >
+                Reserve Rocket
+              </button>
+            )}
+          </div>
+        </div>
```

Comment:
## [Optional Change]

- [ ] React apps are built in Lego blocks comprised of chunks of components as that's the idea behind the library. It would be great to extract this code into a separate component and pass the lopped rocket to it and then receive that as a prop so this component will be lean and clean. Writing React this way helps your codebase structured and not have a huge component that does everything but move some part of it into reusable components. It'll be great if you consider doing it that way 👍🏽